build-tools/packages/build-cli/test/commands/typetests.test.ts

```diff
@@ -2,11 +2,162 @@
  * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
  * Licensed under the MIT License.
  */
-import { assert } from "chai";
+import { assert, expect } from "chai";
+import { PackageJson } from "@fluidframework/build-tools";
 
-import { previousVersion } from "../../src/commands/typetests";
+import {
+	VersionOptions,
+	applyTypeTestVersionOptions,
+	previousVersion,
+	resetBrokenTests,
+} from "../../src/commands/typetests";
+
+/**
+ * A minimal test package.json. It defines only the required fields according to the type definition.
+ */
+function packageMinimal(): PackageJson {
+	return {
+		name: "test-package",
+		version: "2.0.0-internal.6.0.0",
+		scripts: {},
+	};
+}
+
+/**
+ * A test package.json with an enabled typeValidation node.
+ */
+function packageWithTypeValidation(): PackageJson {
+	return {
+		...packageMinimal(),
+		devDependencies: {
+			"test-package-previous": "npm:test-package@2.0.0-internal.4.0.0",
+			"another-dependency": "^1.0.0",
```

Comment:
rather than cluttering up every test here with this extra unrelated dependency, I think it might be better to have a single unit test that unrelated dependencies are left alone, then omit this from the rest of the tests.

Another similar way we can simplify most of the tests by splitting one case out into its own test is with versions.
We could make all the existing tests use a very simple version (even something as trivial as "1" should actually work fine). We can then test elsewhere that versions are handled in a compatible way that works with all the formats. Since this code uses "previousVersion" as to do all version related computations, such tests can be part of "previousVersion"'s test suite (which I think it already has) so this test suite can just use simpler versions.

Reply:
> rather than cluttering up every test here with this extra unrelated dependency, I think it might be better to have a single unit test that unrelated dependencies are left alone, then omit this from the rest of the tests.

Good idea; implemented. I also used simpler versions since, as you pointed out, that logic is already tested independently.